scalafix-core/src/main/scala/scalafix/rewrite/NoAutoTupling.scala

```diff
@@ -0,0 +1,53 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import scalafix.syntax._
+import scala.meta.semantic.Signature
+import scala.collection.mutable.{Set => MutableSet}
+
+case class NoAutoTupling(mirror: Mirror) extends SemanticRewrite(mirror) {
+
+  private[this] val singleTuplePattern = "^Lscala\\/Tuple\\d{1,2}$".r.pattern
+
+  private[this] def addWrappingParens(ctx: RewriteCtx,
+                                      args: Seq[Term.Arg]): Patch =
+    Seq(
+      ctx.addLeft(args.head.tokens.last, "("),
+      ctx.addRight(args.last.tokens.last, ")")
+    ).asPatch
+
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    // "hack" to avoid fixing an argument list more than once due
+    // to recursive matching of multiple parameters lists.
+    val fixed = MutableSet.empty[Seq[Term.Arg]]
```

Comment:
This is OK for now. I think we might be able bake some idempotency into the patch system, because I think this will pop up regularly.